### PR TITLE
Delete profile

### PR DESCRIPTION
**Pull request by mizamidou**

### CODE REVIEW

**Review by stevelw — CHANGES_REQUESTED** [review]
Great work - some things to fix but it's looking good.

**Comment by mizamidou** [comment]
I have done the changes. Please check

File: src/Components/CatProfile.tsx [hunk 1]
```diff
@@ -6,28 +6,33 @@ import FormElement from './Styling/FormElement';
 import Button from './Styling/Button';
 import { Cat } from '../api';
 
+const loggedInUser = 'loggedInUser';
```
Comment (stevelw) [inline src/Components/CatProfile.tsx:9]: Should be `USER1` or some other user in the dev database.

File: src/Components/CatProfile.tsx [hunk 2]
```diff
@@ -6,28 +6,33 @@ import FormElement from './Styling/FormElement';
...
 	const [catName, setCatName] = useState<string>('');
 	const [catPicture, setCatPicture] = useState<string>('');
 	const [catDescription, setCatDescription] = useState<string>('');
 	const [catProfiles, setCatProfiles] = useState<Cat[]>([]);
 	const [isLoading, setIsLoading] = useState<boolean>(true);
+	const [message, setMessage] = useState<string>('');
 
 	useEffect(() => {
-		getCatsProfiles('USER1').then((fetchedProfiles) => {
+		getCatsProfiles('loggedInUser').then((fetchedProfiles) => {
```
Comment (stevelw) [inline src/Components/CatProfile.tsx:20]: This is a string, not the variable from line 9

File: src/Components/CatProfile.tsx [hunk 3]
```diff
@@ -60,22 +65,23 @@ function CatProfile() {
...
 			.catch(() => {
-				console.log('Updates havent been saved');
+				setMessage('Updates havent been saved');
 			});
 	};
 
-	const handleDeleteProfile = (event: React.MouseEvent<HTMLButtonElement>) => {
-		deleteCatProfile('USER1')
+	const handleDeleteProfile = () => {
+		deleteCatProfile('loggedInUser')
```
Comment (stevelw) [inline src/Components/CatProfile.tsx:76]: Same issue

File: src/Components/CatProfile.tsx [hunk 4]
```diff
@@ -84,6 +90,17 @@ function CatProfile() {
 			<h2 className="text-2xl font-bold text-red-950	mb-20">
 				Cat&apos;s Profile
 			</h2>
+			{message && (
+				<div
+					className={`text-2xl font-bold  mb-20
+					${message === 'Updates have been saved' ? 'text-red-950' : ''}
+					${message === 'Updates havent been saved' ? 'text-red-950' : ''}
+					${message === 'Profile Removed' ? 'text-red-950' : ''}
+					${message === 'Failed to delete the profile' ? 'text-red-950' : ''}`}
```
Comment (stevelw) [inline src/Components/CatProfile.tsx:99]: All styling should be in components now. See `src/Components/Styling/FormElement.tsx` for an example.

File: package.json [hunk 5]
```diff
@@ -11,20 +11,24 @@
...
 		"axios": "^1.7.7",
+		"body-parser": "^1.20.3",
+		"dayjs": "^1.11.13",
+		"jsonwebtoken": "^9.0.2",
+		"lodash": "^4.17.21",
 		"mapbox-gl": "^3.8.0",
 		"react": "^18.3.1",
-		"react-dom": "^18.3.1"
+		"react-dom": "^18.3.1",
+		"socket.io": "^4.8.1",
```
Comment (stevelw) [inline package.json:21]: Not sure why these packages are needed? I don't see them used in the code.

**Review by stevelw — CHANGES_REQUESTED** [review]

File: src/Components/CatProfile.tsx (hunk 4 above)
Comment (stevelw) [reply src/Components/CatProfile.tsx:99]: Components are now in #17 awaiting merging to main

File: src/Components/CatProfile.tsx [hunk 6]
```diff
@@ -14,9 +14,10 @@ function CatProfile() {
 	const [isLoading, setIsLoading] = useState<boolean>(true);
 
 	useEffect(() => {
-		getCatsProfiles().then((fetchedProfiles) => {
+		getCatsProfiles('USER1').then((fetchedProfiles) => {
```
[the remark this answers is not in the document]
Comment (stevelw) [reply src/Components/CatProfile.tsx:20]: The version I can see on the PR is using a literal string not the variable.

File: src/Components/CatProfile.tsx [hunk 7]
```diff
@@ -44,7 +41,43 @@ function CatProfile() {
...
+					)
+				);
+			})
+			.catch(() => {
+				console.log('Updates havent been saved');
+			});
+	};
+
+	const handleDeleteProfile = (event: React.MouseEvent<HTMLButtonElement>) => {
+		deleteCatProfile('USER1')
```
[the remark this answers is not in the document]
Comment (stevelw) [reply src/Components/CatProfile.tsx:76]: As with the other one, still showing as a literal string in the PR.

File: src/Components/CatProfile.tsx (hunk 1 above)
Comment (stevelw) [reply src/Components/CatProfile.tsx]: I just checked the current seed data, and we're currently seeding with `user1` in lowercase.

File: src/Components/CatProfile.tsx (hunk 2 above)
Comment (stevelw) [reply src/Components/CatProfile.tsx]: Save case issue as the comment.

File: src/Components/CatProfile.tsx (hunk 3 above)
Comment (stevelw) [reply src/Components/CatProfile.tsx]: Same variable issue as the other comment.

File: src/Components/CatProfile.tsx (hunk 4 above)
Comment (stevelw) [reply src/Components/CatProfile.tsx:99]: Todo in CL-119

File: src/Components/CatProfile.tsx (hunk 1 above)
Comment (stevelw) [reply src/Components/CatProfile.tsx]: The version I can see still says `USER1`?

**Review by stevelw — CHANGES_REQUESTED** [review]
Just the username definition issue with the case

File: src/Components/CatProfile.tsx [hunk 8]
```diff
@@ -1,29 +1,31 @@
 import React from 'react';
 import { useState, useEffect, ChangeEvent } from 'react';
-import { getCatsProfiles } from '../api';
+import { getCatsProfiles, deleteCatProfile, updateCatProfile } from '../api';
 import FormInput from './Styling/FormInput';
 import FormElement from './Styling/FormElement';
 import Button from './Styling/Button';
 import { Cat } from '../api';
 
+const loggedInUser = 'USER1';
```
Comment (jeepies) [inline src/Components/CatProfile.tsx:9]: Remove this line and change it for something like `const { REACT_APP_LOGGED_IN_USER } = process.env`

File: src/Components/CatProfile.tsx [hunk 9]
```diff
@@ -1,29 +1,31 @@
...
 	const [catName, setCatName] = useState<string>('');
 	const [catPicture, setCatPicture] = useState<string>('');
 	const [catDescription, setCatDescription] = useState<string>('');
 	const [catProfiles, setCatProfiles] = useState<Cat[]>([]);
 	const [isLoading, setIsLoading] = useState<boolean>(true);
+	const [message, setMessage] = useState<string>('');
 
 	useEffect(() => {
-		getCatsProfiles().then((fetchedProfiles) => {
+		getCatsProfiles(loggedInUser).then((fetchedProfiles) => {
```
Comment (jeepies) [inline src/Components/CatProfile.tsx:20]: Change the loggedInUser variable to `REACT_APP_LOGGED_IN_USER`

**Review by jeepies — CHANGES_REQUESTED** [review]

**Pull request merged**